wagtail/core/query.py

```diff
@@ -218,25 +220,85 @@ def not_exact_type(self, model):
         """
         return self.exclude(self.exact_type_q(model))
 
-    def public_q(self):
+    def unrestricted_q(self, cache_target=None):
         from wagtail.core.models import PageViewRestriction
 
         q = Q()
-        for restriction in PageViewRestriction.objects.select_related('page').all():
+        for restriction in PageViewRestriction.get_all(cache_target):
             q &= ~self.descendant_of_q(restriction.page, inclusive=True)
         return q
 
-    def public(self):
+    def not_view_restricted(self, cache_target=None):
+        """
+        Filters the queryset to only include pages unaffected by page view restrictions.
+        This differs from ``public()`` in that the result will not include pages
+        affected by restrictions of type ``BaseViewRestriction.NONE``.
```

Comment:
@gasman I think we can safely assume that people aren't using `BaseViewRestriction.NONE` in the wild in this way now - However, since this PR paves the way for more complex configurations, it could technically be used to denote a public section within a private one, as I have done so in this test here: https://github.com/wagtail/wagtail/pull/6300/files#diff-0982e81a156f529061ddfffaa49f59c5f598ba50daf8234340e8d37febff420bR653

If this restriction has special meaning elsewhere though, perhaps we should add something new to allow that to be reflected without fear of it being deleted by accident :)